csrc/transform_replay.cpp

```diff
@@ -240,24 +240,18 @@ TensorDomain* TransformReplay::fullSelfReplay(
 
 void TransformReplay::selfReplay(
     const TensorDomain* self,
-    TensorDomain* new_self) {
+    TensorDomain* new_self,
+    bool ignore_reductions) {
   FUSER_PERF_SCOPE("TransformReplay::selfReplay");
 
-  // NOTE: We could also have reduction IDs involved in transformation that
-  // leads to allocation domain, so technically we should have included
-  // reduction IDs in the replay as well. The reason that we skipped them here
-  // is because this function is used by `RemoveBcastSqueeze`, where we could
-  // have mismatch reduction IDs on the logical between `self` and
-  // `new_self`.
```

Comment:
I added some comments in the header file, which hopefully explain the idea better.